src/encoded/static/components/cart/cart.js

```diff
@@ -1186,6 +1349,66 @@ CartPager.propTypes = {
 };
 
 
+/**
+ * Displays controls at the top of search results within the tab content areas.
+ */
+const CartSearchResultsControls = ({ currentTab, elements, currentPage, totalPageCount, updateCurrentPage }) => {
+    if (currentTab === 'datasets' || totalPageCount > 1) {
+        return (
+            <div className="cart-search-results-controls">
+                {currentTab === 'datasets'
+                    ? <CartRemoveElements elements={elements.map((element) => element['@id'])} />
+                    : <div />}
+                {totalPageCount > 1
+                    ? <CartPager currentPage={currentPage} totalPageCount={totalPageCount} updateCurrentPage={updateCurrentPage} />
+                    : null}
+            </div>
+        );
+    }
+    return null;
+};
+
+CartSearchResultsControls.propTypes = {
+    /** Key of the currently selected tab */
+    currentTab: PropTypes.string.isRequired,
+    /** Array of currently displayed cart items */
+    elements: PropTypes.array.isRequired,
+    /** Zero-based current page to display */
+    currentPage: PropTypes.number.isRequired,
+    /** Total number of pages */
+    totalPageCount: PropTypes.number.isRequired,
+    /** Called when user clicks pager controls */
+    updateCurrentPage: PropTypes.func.isRequired,
+};
+
+
+/**
+ * Add the datasets search results to the given array of datasets.
+ * @param {array} datasets Add new search results to this array
+ * @param {object} currentResults Dataset search results
+ * @return {array} `datasets` with search-result datasets added
+ */
+const addToAccumulatingDatasets = (datasets, currentResults) => {
+    if (currentResults['@graph'] && currentResults['@graph'].length > 0) {
```

Comment:
Storing currentResults['@graph'] may make the code better, albeit computationally it won't make a difference since access to array entry is O(1)